File: src/main/java/com/openlattice/chronicle/services/delete/DataDeletionManager.java
package com.openlattice.chronicle.services.delete;

import com.openlattice.chronicle.data.ChronicleDeleteType;

import java.util.UUID;

/**
 * @author alfoncenzioka &lt;alfonce@openlattice.com&gt;
 */
public interface DataDeletionManager {

    void deleteParticipantAndAllNeighbors(
            UUID organizationId,
            UUID studyId,
            String participantId,
            ChronicleDeleteType chronicleDeleteType,
            String token );

    void deleteStudyAndAllNeighbors(
            UUID organizationId,
            UUID studyId,
            ChronicleDeleteType chronicleDeleteType,
            String token
    );
}

File: src/main/kotlin/com/openlattice/chronicle/services/delete/DataDeletionManager.kt
package com.openlattice.chronicle.services.delete

import com.openlattice.chronicle.data.ChronicleDeleteType
import java.util.*

/**
 * @author alfoncenzioka &lt;alfonce@openlattice.com&gt;
 * @author Matthew Tamayo-Rios &lt;matthew@openlattice.com&gt;
 */
interface DataDeletionManager {
    fun deleteParticipantData(
            organizationId: UUID,
            studyId: UUID,
            participantId: String,
            chronicleDeleteType: ChronicleDeleteType,
    )

    fun deleteStudyData(
            organizationId: UUID,
            studyId: UUID,
            chronicleDeleteType: ChronicleDeleteType,
    )
}
